lib/api/api.js

```diff
@@ -67,6 +72,17 @@ const api = {
             // Pull the source bucket and source object separated by /
             sourceBucket = source.slice(0, slashSeparator);
             sourceObject = source.slice(slashSeparator + 1);
+            sourceVersionId = query ?
+                querystring.parse(query).versionId : undefined;
+            sourceVersionId = sourceVersionId || undefined;
+            if (sourceVersionId && sourceVersionId !== 'null') {
+                try {
+                    sourceVersionId = VID.decrypt(sourceVersionId);
+                } catch (exception) {
+                    return callback(errors.InvalidArgument
+                        .customizeDescription('Invalid version id specified'));
+                }
+            }
```

Comment:
I don't think you have to update the requestContexts for version id since we already parse it from the headers.  @nicolas2bert is adding the additional actions to arsenal for versioning (for instance, s3:GetObjectVersion).  What i'm getting at is that when you build the request context for the "get" part of the object copy, you have to send in the version id on the source since that would not be in the regular place in the header -- it would be in the source header.